src/cpp/session/SessionSourceDatabase.cpp

```diff
@@ -69,6 +69,10 @@ const char * const SourceDocument::SourceDocumentTypeCpp       = "cpp";
 
 namespace {
 
+// cached mapping of document id to document path (facilitates efficient path
+// lookup)
+std::map<std::string, std::string> s_idToPath;
```

Comment:
Yes, my intent in putting it in `SessionSourceDatabase` was to make it reusable. Now centralized here: https://github.com/rstudio/rstudio/commit/dd1b65f26a172afc33d3f6071b85d7f5c1b13e0e